tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,477 @@
+            } catch (WakeupException e) {
+                // OK
+            } finally {
+                consumer.close();
+            }
+        }
+
+        void shutdown() {
+            consumer.wakeup();
+        }
+    }
 
-        TestUtils.waitForCondition(
-            () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
-            "The group did not become empty as expected.");
+    class ConsumerRunnable extends AbstractConsumerRunnable {
+        final String topic;
+        final String groupProtocol;
+        final String strategy;
+        final Optional<String> remoteAssignor;
+
+        public ConsumerRunnable(String broker, String groupId, String groupProtocol, String topic, String strategy,
+                                Optional<String> remoteAssignor, Optional<Properties> customPropsOpt, boolean syncCommit) {
+            super(broker, groupId, customPropsOpt, syncCommit);
+
+            this.topic = topic;
+            this.groupProtocol = groupProtocol;
+            this.strategy = strategy;
+            this.remoteAssignor = remoteAssignor;
+        }
+
+        @Override
+        void configure(Properties props) {
+            super.configure(props);
+            props.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, groupProtocol);
+            if (groupProtocol.toUpperCase(Locale.ROOT).equals(GroupProtocol.CONSUMER.toString())) {
+                remoteAssignor.ifPresent(assignor -> props.put(ConsumerConfig.GROUP_REMOTE_ASSIGNOR_CONFIG, assignor));
+            } else {
+                props.put("partition.assignment.strategy", strategy);
+            }
+        }
+
+        @Override
+        void subscribe() {
+            consumer.subscribe(Collections.singleton(topic));
+        }
+    }
 
-        cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP, "--group", missingGroup};
 
-        ConsumerGroupCommand.ConsumerGroupService service2 = getConsumerGroupService(cgcArgs);
-        Map<String, Throwable> result = service2.deleteGroups();
-        assertTrue(result.size() == 2 &&
-                result.containsKey(GROUP) && result.get(GROUP) == null &&
-                result.containsKey(missingGroup) &&
-                result.get(missingGroup).getMessage().contains(Errors.GROUP_ID_NOT_FOUND.message()),
-            "The consumer group deletion did not work as expected");
+    class AbstractConsumerGroupExecutor {
+        final int numThreads;
+        final ExecutorService executor;
+        final List<AbstractConsumerRunnable> consumers = new ArrayList<>();
+
+        public AbstractConsumerGroupExecutor(int numThreads) {
+            this.numThreads = numThreads;
+            this.executor = Executors.newFixedThreadPool(numThreads);
+        }
+
+        void submit(AbstractConsumerRunnable consumerThread) {
+            consumers.add(consumerThread);
+            executor.submit(consumerThread);
+        }
+
+        void shutdown() {
```

Comment:
We can change this to `close` and make `AbstractConsumerGroupExecutor` extend `Closeable`. That allow us to use `try-with-resources`